custom/src/main/java/com/splunk/opentelemetry/appd/AppdBonusPropagator.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.appd;
+
+import static com.splunk.opentelemetry.appd.AppdBonusConstants.CTX_HEADER_ACCT;
+import static com.splunk.opentelemetry.appd.AppdBonusConstants.CTX_HEADER_APP;
+import static com.splunk.opentelemetry.appd.AppdBonusConstants.CTX_HEADER_BT;
+import static com.splunk.opentelemetry.appd.AppdBonusConstants.CTX_HEADER_ENV;
+import static com.splunk.opentelemetry.appd.AppdBonusConstants.CTX_HEADER_SERVICE;
+import static com.splunk.opentelemetry.appd.AppdBonusConstants.CTX_HEADER_TIER;
+import static com.splunk.opentelemetry.appd.AppdBonusConstants.CTX_KEY;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.ContextKey;
+import io.opentelemetry.context.propagation.TextMapGetter;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import java.util.Arrays;
+import java.util.Collection;
+import javax.annotation.Nullable;
+
+final class AppdBonusPropagator implements TextMapPropagator {
+
+  public static final ContextKey<AppdBonusContext> CONTEXT_KEY = ContextKey.named(CTX_KEY);
+  private static final AppdBonusPropagator INSTANCE = new AppdBonusPropagator();
+
+  @Nullable private String serviceName;
+  @Nullable private String environmentName;
+
+  @VisibleForTesting
+  AppdBonusPropagator() {}
```

Comment:
at first glance couldn't see it used in the test (could be wrong)

Reply:
It's used in the `AppdBonusPropagatorTest`. :)